packages/bitfield/tests/lib/BitField.test.ts

```diff
@@ -0,0 +1,355 @@
+import { BitField } from '../../src';
+
+describe('BitField', () => {
+	const NumberFlags = {
+		Read: 1,
+		Write: 2,
+		Edit: 4,
+		Delete: 8
+	} as const;
+
+	const BigIntFlags = {
```

Comment:
Can you make sure enums work too? 🙏

Reply:
I don't know if enums will work, or if I can map their types nicely, but it can be implemented in a future PR.

2-way enums, that is. 1-way ones should be fine.

Reply:
1. Check out how we did it in shapeshift's nativeenum

Reply:
The biggest issue for me is mapping the values correctly while ignoring the reverse maps, both in the types and in the data.

Chances are high I'll just make a function to transform 2-way enums into 1-way ones, otherwise, if done within the constructor, the types could get potentially A LOT more complex, and it's already nightmare-ish as-is.